### PR TITLE
DOCSP-41137: Count Documents

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41137>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41137-count-docs/read/count/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read.txt [hunk 1]
```diff
@@ -88,8 +81,8 @@ The following example returns the number of documents in the specified collectio
    :copyable:
    :dedent:
 
-.. TODO: To learn more about the ``count_documents()`` method, see the
-.. :ref:`kotlin-sync-accurate-count` guide.
+To learn more about the ``countDocuments()`` method, see the
+:ref:`kotlin-sync-accurate-count` guide.
```
Comment (norareidy) [inline source/read.txt:85]: S: since you're linking to a section within the Count Documents guide, I'd reword to something like:
```suggestion
:ref:`kotlin-sync-accurate-count` section of the Count Documents guide.
```

Comment (norareidy) [reply source/read.txt:85]: Applies to the two links below as well

File: source/read/count.txt [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+   :keywords: number, amount, estimation, code example
+
+Overview
+---------
+
+In this guide, you can learn how to retrieve accurate and estimated counts of the
+number of documents in a collection.
+
+.. _kotlin-sync-accurate-count:
+
```
Comment (norareidy) [inline source/read/count.txt:27]: S: not necessary, but it might be helpful to use one of the Atlas sample datasets for the code examples on this page and introduce it here (similar to the [C++ count documents](https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40801-count/read/count/) page). This could help make the current code examples a little more specific and would provide an output

File: source/read/count.txt [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+   * - ``maxTime``
+     - | The maximum amount of time to allow the operation to run, in milliseconds.
+
+   * - ``collection``
+     - | An instance of ``Collation``.
+
+   * - ``hint``
+     - | Gets or sets the index to scan for documents.
+
```
Comment (norareidy) [inline source/read/count.txt:91]: S: I think it would be helpful to add a code example here that passes an options argument

File: source/read/count.txt [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+     - | The number of documents to skip before returning results.
+
+   * - ``limit``
+     - | The maximum number of documents to count. Must be a positive integer.
+
+   * - ``maxTime``
+     - | The maximum amount of time to allow the operation to run, in milliseconds.
+
+   * - ``collection``
+     - | An instance of ``Collation``.
```
Comment (norareidy) [inline source/read/count.txt:87]: Q: should this say "collation" instead of "collection"? Also, you could make the description a little more specific:
```suggestion
   * - ``collection``
     - | The collation to use for the operation.
```

File: source/read/count.txt [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+.. code-block:: kotlin
+   
+   collection.countDocuments(eq("author", "Leslie"))
+
+Customize Count Behavior
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``countDocuments()`` method accepts optional parameters in the form of a
+``CountOptions`` object, which represents options you can use to configure the count
```
Comment (norareidy) [inline source/read/count.txt:61]: Q: how do you set these options - by specifying fields of a `CountOptions` object? Might be good to clarify that

File: source/read/count.txt [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+   * - ``hint``
+     - | Gets or sets the index to scan for documents.
+
+.. _kotlin-sync-estimated-count:
+
+Retrieve an Estimated Count
+---------------------------
+
+You can get an estimate of the number of documents in a collection by calling
+the ``estimatedDocumentCount()`` method. The method estimates the amount of
```
Comment (norareidy) [inline source/read/count.txt:98]: S: to match the syntax of the accurate count section:
```suggestion
Use the ``estimatedDocumentCount()`` method to retrieve an estimate of the number of documents in a collection. The method estimates the amount of
```

File: source/read/count.txt [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+operation. If you don't specify any options, the driver does not customize the count
+operation.
+
+The following table describes the options you can set to customize ``countDocuments()``:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
```
Comment (norareidy) [inline source/read/count.txt:71]: S: "Field" or "Option" might be better words than "Property" here

**Review by norareidy — CHANGES_REQUESTED** [review]
Nice work! Added some suggestions / questions

File: source/read/count.txt (hunk 5 above)
Comment (mcmorisi) [reply source/read/count.txt:61]: Can add some clarifying wording.

File: source/read/count.txt [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+       :visible: false
+
+       21349
+
+Count Specific Documents
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To return a count of documents that match specific search criteria, specify your query
+in the ``countDocuments()`` method. The following example prints a count of all documents
+in the ``movies`` collection with a ``year`` field value equal to ``1930``:
```
Comment (norareidy) [inline source/read/count.txt:68]: S: to avoid "with":
```suggestion
in the ``movies`` collection that have a ``year`` field value equal to ``1930``:
```

**Review by norareidy — APPROVED** [review]
LGTM + one small suggestion
